x-pack/legacy/plugins/ml/public/data_frame/pages/transform_management/components/transform_list/transform_list.tsx

```diff
@@ -76,6 +90,75 @@ export const DataFrameTransformList: SFC = () => {
   // Call useRefreshInterval() after the subscription above is set up.
   useRefreshInterval(setBlockRefresh);
 
+  const onQueryChange = ({ query }: { query: Query }) => {
+    setIsLoading(true);
+    const clauses = query.ast.clauses;
+
+    if (clauses.length === 0) {
+      setFilteredTransforms(transforms);
+      setIsLoading(false);
+      return;
+    }
+    // keep count of the number of matches we make as we're looping over the clauses
+    // we only want to return transforms which match all clauses, i.e. each search term is ANDed
+    // { transform-one:  { transform: { id: transform-one, config: {}, state: {}, ... }, count: 0 }, transform-two: {...} }
+    const matches: { [key: string]: any } = transforms.reduce((p: { [key: string]: any }, c) => {
```

Comment:
rather than `{ [key: string]: any }`, `Record<string, any>` might be cleaner.

Reply:
Updated in 413a53b73f955c04e3c5319d14dc4a30a3d09b5d